impl11/ddraw/SteamVR.cpp

```diff
@@ -298,96 +299,81 @@ void GetSteamVRPositionalData(float* yaw, float* pitch, float* roll, float* x, f
-			x_cockpitlook = (float)PlayerDataTable[*g_playerIndex].cockpitXReference / g_fXWAUnitsToMetersScale;
-			y_cockpitlook = (float)PlayerDataTable[*g_playerIndex].cockpitYReference / g_fXWAUnitsToMetersScale;
-			z_cockpitlook = (float)PlayerDataTable[*g_playerIndex].cockpitZReference / g_fXWAUnitsToMetersScale;
-
-
-			// Build a transform matrix to "undo" the CockpitLook transformation based on HMD pose estimated in the previous frame
-			Matrix4 rotMatrixYaw, rotMatrixPitch, posMatrix;
-			rotMatrixYaw.identity();	rotMatrixYaw.rotateY(yaw_cockpitlook);
-			rotMatrixPitch.identity();	rotMatrixPitch.rotateX(-pitch_cockpitlook);
-			posMatrix.identity();		posMatrix.translate(-x_cockpitlook, -y_cockpitlook, -z_cockpitlook);			
-			// Create inverse matrix by applying the opposite angles, in inverse order
-			m4_UndoCockpitLook = posMatrix * rotMatrixYaw * rotMatrixPitch;
-
 			Matrix4 m4_hmdPose = HmdMatrix34toMatrix4(trackedDevicePose.mDeviceToAbsoluteTracking);
 
-			// Recommended settings in CockpitLook.cfg to avoid jitter due to cockpit shake
-			// positional_x_multiplier = 0
-			// positional_y_multiplier = 0
-			// positional_z_multiplier = 0
-			// 
-			// DEBUG: you can cancel positional tracking completely by uncommenting these lines
-			// g_fPosXMultiplier = 0;
-			// g_fPosYMultiplier = 0;
-			// g_fPosZMultiplier = 0;
-
-			// Invert axes so that it works properly. I don't understand yet why this is necessary.
-			int col = 3; //4th column (index 3) of the pose matrix is the translation vector x,y,z
-			m4_hmdPose[col * 4] =		-m4_hmdPose[col * 4]	* g_fPosXMultiplier;
-			m4_hmdPose[col * 4 + 1] =	-m4_hmdPose[col * 4 + 1]* g_fPosYMultiplier;
-			m4_hmdPose[col * 4 + 2] =	-m4_hmdPose[col * 4 + 2]* g_fPosZMultiplier;
-
-			//Compose it with the transformation matrix for the actual HMD pose in the current frame (this is the correct view space for rendering, including roll)
-			Matrix4 m4_correctionMatrix = m4_hmdPose * m4_UndoCockpitLook;
-			Matrix4toHmdMatrix34(m4_correctionMatrix, m34_correctionMatrix);  // This matrix contains all positional and rotational data.
-			
+			if (g_bCorrectedHeadTracking) {
+				// We need to apply the perspective correction between the pose predicted in the previous frame
+				// (used by CockpitLook in this frame and implicitly in the 2D->3D retroprojection)
+				// and the actual pose just returned by WaitGetPoses() that will be used to do the D3D render
+
+				// Obtain the yaw, pitch that was applied by CockpitLook. Roll was ignored.
+				float yaw_cockpitlook, pitch_cockpitlook, x_cockpitlook, y_cockpitlook, z_cockpitlook;
+				Matrix4 m4_UndoCockpitLook;
+				yaw_cockpitlook = (float)(PlayerDataTable[*g_playerIndex].cockpitCameraYaw * 360.0f / 65535.0f);
+				pitch_cockpitlook = (float)(PlayerDataTable[*g_playerIndex].cockpitCameraPitch * 360.0f / 65535.0f);
```

Comment:
This is not terribly important, but we could declare a new constant and set it to 360.0f / 65535.0f, then we can use that constant here to save a few processor cycles.